v8.go

```diff
@@ -77,14 +77,19 @@ type Snapshot struct{ data C.StartupData }
 
 func newSnapshot(data C.StartupData) *Snapshot {
 	s := &Snapshot{data}
-	runtime.SetFinalizer(s, (*Snapshot).release)
```

Comment:
The finalizer will run when the context (and the isolate) are garbage collected.  That doesn't necessarily happen when `ctx=nil` is run.  In fact, the go runtime doesn't guarantee that finalizers will ever run -- it may be that the runtime decides that it doesn't need to GC anything.  You can force the GC to run using runtime.GC(), but I don't know whether that forces finalizers to run.  It's worth testing.